devtools/gradle/src/main/java/io/quarkus/gradle/dependency/ConditionalDependenciesEnabler.java

```diff
@@ -0,0 +1,193 @@
+        for (ExtensionDependency extension : extensions) {
+            for (Dependency dependency : extension.conditionalDependencies) {
+                possibleVariant.put(DependencyUtils.asFeatureName(dependency), extension);
+            }
+        }
+        return possibleVariant;
+    }
+
+    private void resolveConditionalDependencies(DependencyHandler dependencies,
+            List<ExtensionDependency> extensions, Configuration existingDependencies) {
+        final Configuration conditionalDeps = createConditionalDependenciesConfiguration(project, existingDependencies,
+                extensions);
+
+        Set<ResolvedArtifact> runtimeArtifacts = existingDependencies.getResolvedConfiguration().getResolvedArtifacts();
+        List<ExtensionDependency> validConditionalDependencies = new ArrayList<>();
+
+        for (ResolvedArtifact artifact : conditionalDeps.getResolvedConfiguration().getResolvedArtifacts()) {
+            ExtensionDependency extensionDependency = getExtensionInfoOrNull(artifact, artifact.getFile(), dependencies);
+            if (extensionDependency != null) {
+                if (extensionDependency.dependencyCondition != null) {
+                    if (DependencyUtils.exists(runtimeArtifacts, extensionDependency.dependencyCondition)) {
+                        enableConditionalDependency(extensionDependency.extensionId);
+                        validConditionalDependencies.add(extensionDependency);
+                        if (!extensionDependency.conditionalDependencies.isEmpty()) {
+                            featureVariants.putAll(extractFeatureVariants(Collections.singletonList(extensionDependency)));
+                        }
+                    }
+                }
+            }
+        }
+
+        Configuration enhancedDependencies = DependencyUtils.duplicateConfiguration(project,
+                project.getConfigurations().getByName(JavaPlugin.IMPLEMENTATION_CONFIGURATION_NAME));
+
+        List<ExtensionDependency> newConditionalDependencies = new ArrayList<>();
+
+        // The dependency may have been excluded, if no, no need to look for transitive dependencies
+        Set<ResolvedArtifact> resolvedArtifacts = enhancedDependencies.getResolvedConfiguration()
+                .getResolvedArtifacts();
+
+        for (ExtensionDependency extension : validConditionalDependencies) {
+            if (DependencyUtils.exists(resolvedArtifacts, extension.asDependency(dependencies))
+                    && !extension.conditionalDependencies.isEmpty()) {
+                newConditionalDependencies.add(extension);
+            }
+        }
+
+        if (!newConditionalDependencies.isEmpty()) {
+            resolveConditionalDependencies(project.getDependencies(),
+                    newConditionalDependencies, enhancedDependencies);
+        }
+    }
+
+    private Configuration createConditionalDependenciesConfiguration(Project project, Configuration existingDeps,
```

Comment:
The `project` is an instance variable, right? I.e. it doesn't have to be passed in as an arg?

Reply:
right!